src/xenia/kernel/xam/xam_user.cc

```diff
@@ -65,6 +65,28 @@ X_HRESULT_result_t XamUserGetXUID_entry(dword_t user_index, dword_t type_mask,
 }
 DECLARE_XAM_EXPORT1(XamUserGetXUID, kUserProfiles, kImplemented);
 
+dword_result_t XamUserGetIndexFromXUID_entry(qword_t xuid, dword_t flags,
+                                             pointer_t<uint32_t> index) {
+  if (!index) {
+    return X_E_INVALIDARG;
+  }
+
+  auto profile_manager = kernel_state()->xam_state()->profile_manager();
+  const uint8_t user_index =
+      profile_manager->GetUserIndexAssignedToProfile(xuid);
+  if (user_index == XUserIndexAny) {
+    return X_E_NO_SUCH_USER;
+  }
+
+  if (user_index < XUserMaxUserCount) {
```

Comment:
You don't need this check. GetUserIndexAssignedToProfile actively prevents returning user_index bigger than 3

Reply:
got it